lib/utils.pm

```diff
@@ -761,7 +761,10 @@ sub handle_login {
             select_user_gnome($myuser);
         }
     }
-    assert_screen 'displaymanager-password-prompt', no_wait => 1;
+    if (!check_screen('displaymanager-password-prompt', 30)) {
```

Comment:
I wonder why not to use multitag assert here or that's not possible to catch unfocused password prompt?
Advantage will be that if password prompt is unfocused, we won't wait for 30 seconds before selecting it.

Reply:
+1 what @rwx788 says, the `check_screen` is a no-go for me here

Reply:
Yes, it is not possible.

As I stated here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6596#discussion_r249836457

And I explained it here: https://progress.opensuse.org/issues/46223#note-6

Comparing this job where it worked with a failing one:
- Passed: https://openqa.opensuse.org/tests/832459#step/user_gui_login/5
with
- Failing: https://openqa.opensuse.org/tests/831695#step/user_gui_login/6

There is no difference between a focused textbox from a non-focused textbox. If you try to verify if it is not focused, you get false positives, since there is a 50% chance (it really depends on the blinking frecuency, but you get the point) of getting an focused textbox without cursor.
The only thing we can be sure is that if we see the cursor, it is focused, and to avoid the test to fail, I need check_screen.

Reply:
But the two screens look quite different. Maybe only very subtle a one-pixel border around the entryfield which looks like black-on-black but the other logos are bright in one case and dim in the other case. Can't we use that? If it is *really* not possible to detect if the password field is unfocussed then this is a pretty messed up bug in sddm that should definitely be reported and fixed by adjusting the design.

Reply:
ok, I can agree with an UX bug. There should be a unambiguous difference between a focused and an unfocused textbox.

Creating the bug and adapting the test.

Reply:
That should be most likely an upstream bug. Have you looked on bugs.kde.org if maybe there is one already? But honestly I am not yet sure if that is really the problem. As I stated: IMHO the two screens *do* look quite different

Reply:
I didn't.

I have found a list of bugs that could be related. bugs.kde.org redirected me to https://github.com/sddm/sddm for sddm bugs.

I put here the list to carefully check them tomorrow (I just did a quick look):

- https://github.com/sddm/sddm/issues/990 (theme problem?)
- https://github.com/sddm/sddm/issues/859 (No focus and unresponsive keyboard until click with dual monitor)
- https://github.com/sddm/sddm/issues/612 (Focus is not on password field after boot)
- https://github.com/sddm/sddm/issues/363 (visual feedback problems)

At least openQA and in my monitors I don't see any difference in the textbox. I will try to use it manually to see if there is another element in the screen that can give a hint about a focused password textbox

Reply:
Does

![focussed_unfocussed](https://user-images.githubusercontent.com/1693432/51698055-94843e80-2009-11e9-849c-ed9bc1699934.png)

convince you? The upper half is "focussed", the lower is "unfocussed", e.g. by manually selecting the key below the textfield. The "focussed" field has a blurry blue-ish border